nexus/db-queries/src/db/datastore/scim_provider_store.rs

```diff
@@ -7,124 +7,1791 @@
+                    e
+                } else {
+                    ProviderStoreError::StoreError(e.into())
+                }
+            })?;
+
+        Ok(user)
     }
 
     async fn delete_user_by_id(
         &self,
-        _user_id: &str,
+        user_id: &str,
     ) -> Result<ProviderStoreDeleteResult, ProviderStoreError> {
-        return Err(ProviderStoreError::StoreError(anyhow!(
-            "not implemented!"
-        )));
+        let user_id: SiloUserUuid = match user_id.parse() {
+            Ok(v) => v,
+            Err(_) => {
+                return Err(ProviderStoreError::StoreError(anyhow!(
+                    "user id must be uuid"
+                )));
+            }
+        };
+
+        let authz_silo_user = authz::SiloUser::new(
+            self.authz_silo.clone(),
+            user_id,
+            LookupType::by_id(user_id),
+        );
+        self.opctx
+            .authorize(authz::Action::Delete, &authz_silo_user)
+            .await
+            .map_err(external_error_to_provider_error)?;
+        let conn = self
+            .datastore
+            .pool_connection_authorized(self.opctx)
+            .await
+            .map_err(|err| {
+                ProviderStoreError::StoreError(anyhow!(
+                    "Failed to access DB connection: {err}"
+                ))
+            })?;
+
+        let err: OptionalError<ProviderStoreError> = OptionalError::new();
+
+        let deleted_user = self
+            .datastore
+            .transaction_retry_wrapper("scim_delete_user_by_id")
+            .transaction(&conn, |conn| {
+                let err = err.clone();
+
+                async move {
+                    self.delete_user_by_id_in_txn(&conn, err, user_id).await
+                }
+            })
+            .await
+            .map_err(|e| {
+                if let Some(e) = err.take() {
+                    e
+                } else {
+                    ProviderStoreError::StoreError(e.into())
+                }
+            })?;
+
+        if deleted_user {
+            Ok(ProviderStoreDeleteResult::Deleted)
+        } else {
+            Ok(ProviderStoreDeleteResult::NotFound)
+        }
     }
 
     async fn get_group_by_id(
         &self,
-        _group_id: &str,
+        group_id: &str,
     ) -> Result<Option<StoredParts<Group>>, ProviderStoreError> {
-        return Err(ProviderStoreError::StoreError(anyhow!(
-            "not implemented!"
-        )));
+        let group_id: SiloGroupUuid = match group_id.parse() {
+            Ok(v) => v,
+            Err(_) => {
+                return Err(ProviderStoreError::StoreError(anyhow!(
+                    "group id must be uuid"
+                )));
+            }
+        };
+
+        self.opctx
+            .authorize(
+                authz::Action::ListChildren,
+                &SiloGroupList::new(self.authz_silo.clone()),
+            )
+            .await
+            .map_err(external_error_to_provider_error)?;
```

Comment:
Same as user, this should probably check read on the user, not list children on the silo.

https://gist.github.com/david-crespo/afdb4671a7f72126accf19a7eb8f9319#authorization-semantics